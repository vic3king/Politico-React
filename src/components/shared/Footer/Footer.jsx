import React from 'react';
import '../../../style/footer.scss';

const Footer = () => {
  return (
    <React.Fragment>
      <footer className="main-footer">
        <nav>
          <ul className="main-footer__links">
            <li className="main-footer__link">
              <a href="/">Support</a>
            </li>
            <li className="main-footer__link">
              <a href="/">Terms of Use</a>
            </li>
          </ul>
        </nav>
      </footer>
    </React.Fragment>
  );
};

export default Footer;
