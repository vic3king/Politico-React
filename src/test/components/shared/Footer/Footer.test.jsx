import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../../../../components/shared/Footer/Footer';

describe('Footer component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a footer tag', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.find('footer'));
    expect(wrapper.hasClass('main-footer'));
  });
});
