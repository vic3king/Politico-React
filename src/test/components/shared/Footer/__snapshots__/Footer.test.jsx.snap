// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <footer
        className="main-footer"
      >
        <nav>
          <ul
            className="main-footer__links"
          >
            <li
              className="main-footer__link"
            >
              <a
                href="/"
              >
                Support
              </a>
            </li>
            <li
              className="main-footer__link"
            >
              <a
                href="/"
              >
                Terms of Use
              </a>
            </li>
          </ul>
        </nav>
      </footer>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <ul
            className="main-footer__links"
          >
            <li
              className="main-footer__link"
            >
              <a
                href="/"
              >
                Support
              </a>
            </li>
            <li
              className="main-footer__link"
            >
              <a
                href="/"
              >
                Terms of Use
              </a>
            </li>
          </ul>
        </nav>,
        "className": "main-footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="main-footer__links"
          >
            <li
              className="main-footer__link"
            >
              <a
                href="/"
              >
                Support
              </a>
            </li>
            <li
              className="main-footer__link"
            >
              <a
                href="/"
              >
                Terms of Use
              </a>
            </li>
          </ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Support
                </a>
              </li>,
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Terms of Use
                </a>
              </li>,
            ],
            "className": "main-footer__links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="/"
                >
                  Support
                </a>,
                "className": "main-footer__link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Support",
                  "href": "/",
                },
                "ref": null,
                "rendered": "Support",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="/"
                >
                  Terms of Use
                </a>,
                "className": "main-footer__link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Terms of Use",
                  "href": "/",
                },
                "ref": null,
                "rendered": "Terms of Use",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
      "type": "footer",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <footer
          className="main-footer"
        >
          <nav>
            <ul
              className="main-footer__links"
            >
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Support
                </a>
              </li>
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Terms of Use
                </a>
              </li>
            </ul>
          </nav>
        </footer>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <nav>
            <ul
              className="main-footer__links"
            >
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Support
                </a>
              </li>
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Terms of Use
                </a>
              </li>
            </ul>
          </nav>,
          "className": "main-footer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="main-footer__links"
            >
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Support
                </a>
              </li>
              <li
                className="main-footer__link"
              >
                <a
                  href="/"
                >
                  Terms of Use
                </a>
              </li>
            </ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="main-footer__link"
                >
                  <a
                    href="/"
                  >
                    Support
                  </a>
                </li>,
                <li
                  className="main-footer__link"
                >
                  <a
                    href="/"
                  >
                    Terms of Use
                  </a>
                </li>,
              ],
              "className": "main-footer__links",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="/"
                  >
                    Support
                  </a>,
                  "className": "main-footer__link",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Support",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": "Support",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="/"
                  >
                    Terms of Use
                  </a>,
                  "className": "main-footer__link",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Terms of Use",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": "Terms of Use",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "nav",
        },
        "type": "footer",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
